app/grants/views.py

```diff
@@ -1704,3 +1708,47 @@ def toggle_grant_favorite(request, grant_id):
     return JsonResponse({
         'action': 'follow'
     })
+
+
+@login_required
+def verify_grant(request, grant_id):
+    grant = Grant.objects.get(pk=grant_id)
+
+    if grant.twitter_verified:
+        return JsonResponse({
+            'ok': True,
+            'msg': 'Grant was verified previously'
+        })
+
+    auth = tweepy.OAuthHandler(TWITTER_CONSUMER_KEY, TWITTER_CONSUMER_SECRET)
+    auth.set_access_token(TWITTER_ACCESS_TOKEN, TWITTER_ACCESS_SECRET)
+    try:
+        api = tweepy.API(auth)
+        last_tweet = api.user_timeline(screen_name=grant.twitter_handle_1, count=1, tweet_mode="extended")
+    except tweepy.TweepError as e:
+        return JsonResponse({
+            'ok': False,
+            'msg': 'Bad twitter credential'
+        })
+
+    user_code = get_user_code(request.user.profile.id, emoji_codes)
+    text = f"I am verifying my ownership of the { grant.title }"
+
+    has_code = user_code in last_tweet.full_text
+    has_text = text in last_tweet.full_text
+
+    if has_code and has_text:
```

Comment:
noob question, this checks the username of the twitter account + grant id, so that i cant do a URL param attack on the endpoint.. right?

Reply:
Yep, it verifies for twitter account + grant id + is member of this grant + the generated emoji code corresponds to the given grant and users that is verifying.